src/pilot/assign_role.py

```diff
@@ -274,26 +274,26 @@ def define_target_raid_config(role, drac_client):
 def get_raid_controller_id(drac_client):
     disk_ctrls = drac_client.list_raid_controllers()
 
-    raid_controller_ids = [
-        c.id for c in disk_ctrls if drac_client.is_raid_controller(c.id)]
+    raid_controller_ids = []
 
-    number_raid_controllers = len(raid_controller_ids)
+    for cnt in disk_ctrls:
+        if drac_client.is_raid_controller(cnt.id):
```

Comment:
I don't think this code will quite work the way you want.  is_raid_controller() will return True for both PERC RAID controllers and BOSS controllers.  See: https://github.com/openstack/python-dracclient/blob/master/dracclient/resources/raid.py#L522
This means that the line above will add the BOSS controller to the end of the list, and the elif will never execute.  If you want the BOSS controllers to be at the beginning of the list, then check to see if the controller is a BOSS first and insert it at the beginning else if it's a RAID controller then add it at the end.  In other words reverses this code so the BOSS check is handled in the "if" and the RAID check is handled in the "elif"

Reply:
Yes agreed Chris. I have removed line no. 282 and 283.